Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1335,7 +1335,38 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track
     Vector<GridTrack>& allTracks = tracks(m_direction);
 
     while (auto* gridItem = iterator.nextGridItem()) {
+
         bool isNewEntry = itemsSet.add(gridItem).isNewEntry;
+
+        // Masonry Track Sizing
+        // https://drafts.csswg.org/css-grid-3/#track-sizing
+        // We should only compute track sizes on a subset of items.
+        //
+        // - Items placed at the first implicit line in the masonry axis.
+        // - Items that have a specified definite placement in the grid axis.
+        // - Items that span all grid axis tracks.
+        if (m_renderGrid->areMasonryRows() || m_renderGrid->areMasonryColumns()) {
```

Comment:
Since we've always skipped sizing for a masonry axis, this check if functionally checking if the 'other' axis is masonry.

Might be nice as `m_renderGrid->isMasonry(m_direction == ForRows ? ForColumns : ForRows)`